src/main/java/dev/mccue/resolve/core/CacheHandler.java

```diff
@@ -0,0 +1,84 @@
+package dev.mccue.resolve.core;
+
+import java.io.FileOutputStream;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Locale;
+
+public final class CacheHandler {
+    public enum OSLabel {
+        Linux, MacOS, Windows
+    }
+    private static OSLabel currentOS;
+    private static String CACHE_PATH;
+
+    public CacheHandler() {
+        /* This only handles the default cache path and will need to
+        *  be modified when new features are implemented (e.g. CLI --cache option)
+        */
+        checkOS();
+        switch (currentOS){
+            case Linux:
+                CACHE_PATH = System.getProperty("user.home") + "/.cache/jresolve";
+                break;
+            case MacOS:
+                CACHE_PATH = System.getProperty("user.home") + "/Library/Caches/JResolve";
+                break;
+            case Windows:
+                CACHE_PATH = System.getenv("APPDATA") + "\\JResolve\\Cache";
+                break;
+            case null:
+                throw new IllegalArgumentException("Operating System could not be determined.");
+            case default:
+                break;
+        }
+    }
```

Comment:
You've written a race condition here and I'm not really sure why.

If two threads initialize CacheHandler in parallel, you will race to initialize the CACHE_PATH String. This may or may not be undefined behavior - I would have to check the JLS spec and the internals of String - if all fields are final all the way down it might be safe.

It is a lot simpler and raises fewer hard questions to just initialize these as static final things.

```java
private static final OSLabel CURRENT_OS;
private static final OSLabel CACHE_PATH;

static {
    String operatingSystem = System.getProperty("os.name", "generic").toLowerCase(Locale.ENGLISH);
    OSLabel currentOS;
    if (operatingSystem.contains("nux")) {
        currentOS = OSLabel.Linux;
    } else if (operatingSystem.contains("mac")) {
        currentOS = OSLabel.MacOS;
    } else if (operatingSystem.contains("win")) {
        currentOS = OSLabel.Windows;
    }
    
    CURRENT_OS = currentOs;
    
    CACHE_PATH = switch (currentOS){
            case Linux ->
                System.getProperty("user.home") + "/.cache/jresolve";
            case MacOS ->
                System.getProperty("user.home") + "/Library/Caches/JResolve";
            case Windows ->
                System.getenv("APPDATA") + "\\JResolve\\Cache";
            case null ->
                throw new IllegalStateException("Operating System could not be determined.");
        }
}